src/js/tracks/text-track.js

```diff
@@ -34,288 +100,223 @@ import XHR from 'xhr';
-  let label = options['label'] || '';
-  let language = options['language'] || options['srclang'] || '';
-  let id = options['id'] || 'vjs_text_track_' + Guid.newGUID();
+    if (browser.IS_IE8) {
+      tt = document.createElement('custom');
 
-  if (kind === 'metadata' || kind === 'chapters') {
-    mode = 'hidden';
-  }
+      for (let prop in TextTrack.prototype) {
+        if (prop !== 'constructor') {
+          tt[prop] = TextTrack.prototype[prop];
+        }
+      }
+    }
 
-  tt.cues_ = [];
-  tt.activeCues_ = [];
+    tt.tech_ = options.tech;
 
-  let cues = new TextTrackCueList(tt.cues_);
-  let activeCues = new TextTrackCueList(tt.activeCues_);
+    let mode = TextTrackEnum.TextTrackMode[options.mode] || 'disabled';
+    let kind = TextTrackEnum.TextTrackKind[options.kind] || 'subtitles';
+    let label = options.label || '';
+    let language = options.language || options.srclang || '';
+    let id = options.id || 'vjs_text_track_' + Guid.newGUID();
 
-  let changed = false;
-  let timeupdateHandler = Fn.bind(tt, function() {
-    this['activeCues'];
-    if (changed) {
-      this['trigger']('cuechange');
-      changed = false;
+    if (kind === 'metadata' || kind === 'chapters') {
+      mode = 'hidden';
     }
-  });
-  if (mode !== 'disabled') {
-    tt.tech_.on('timeupdate', timeupdateHandler);
-  }
 
-  Object.defineProperty(tt, 'kind', {
-    get: function() {
-      return kind;
-    },
-    set: Function.prototype
-  });
-
-  Object.defineProperty(tt, 'label', {
-    get: function() {
-      return label;
-    },
-    set: Function.prototype
-  });
-
-  Object.defineProperty(tt, 'language', {
-    get: function() {
-      return language;
-    },
-    set: Function.prototype
-  });
-
-  Object.defineProperty(tt, 'id', {
-    get: function() {
-      return id;
-    },
-    set: Function.prototype
-  });
-
-  Object.defineProperty(tt, 'mode', {
-    get: function() {
-      return mode;
-    },
-    set: function(newMode) {
-      if (!TextTrackEnum.TextTrackMode[newMode]) {
-        return;
-      }
-      mode = newMode;
-      if (mode === 'showing') {
-        this.tech_.on('timeupdate', timeupdateHandler);
+    tt.cues_ = [];
+    tt.activeCues_ = [];
+
+    let cues = new TextTrackCueList(tt.cues_);
+    let activeCues = new TextTrackCueList(tt.activeCues_);
+    let changed = false;
+    let timeupdateHandler = Fn.bind(tt, function() {
+      this.activeCues;
+      if (changed) {
+        this.trigger('cuechange');
+        changed = false;
       }
-      this.trigger('modechange');
-    }
-  });
+    });
 
-  Object.defineProperty(tt, 'cues', {
-    get: function() {
-      if (!this.loaded_) {
-        return null;
-      }
+    if (mode !== 'disabled') {
+      tt.tech_.on('timeupdate', timeupdateHandler);
+    }
 
-      return cues;
-    },
-    set: Function.prototype
-  });
+    Object.defineProperty(tt, 'kind', {
+      get() {
+        return kind;
+      },
+      set: () => {}
```

Comment:
I'm in favor of sticking with the shorthand syntax for consistency (but good to know it gets optimized in the transpiling).